lib/cocoapods/installer/xcode/pods_project_generator/target_installer_helper.rb

```diff
@@ -0,0 +1,65 @@
+module Pod
+  class Installer
+    class Xcode
+      class PodsProjectGenerator
+        class TargetInstallerHelper
+          # @param [Generator] generator
+          #        the generator to use for generating the content.
+          #
+          # @param [Pathname] path
+          #        the pathname to save the content into.
+          #
+          # Saves the content the provided path unless the path exists and the contents are exactly the same.
+          #
+          def self.update_changed_file(generator, path)
```

Comment:
we can use `module_function` instead of explicitly declaring these as `self.`, and then include the module in the 2 classes that use it